odoo/addons/base/models/res_users.py

```diff
@@ -662,11 +669,7 @@ def context_get(self):
         }
 
         # ensure the language is set and is compatible with the web client
-        lang = context.get('lang') or (request and request.default_lang()) or DEFAULT_LANG
-        if lang == 'ar_AR':
-            context['lang'] = 'ar'
-        if lang in babel.core.LOCALE_ALIASES:
-            context['lang'] = babel.core.LOCALE_ALIASES[lang]
```

Comment:
2 small questions.
1. I think `context['lang'] = 'ar'` should be `lang = 'ar'`
https://github.com/odoo/odoo/blob/adf70bf9dcb2786e7cd00e3f84e0dc933161bc4b/odoo/http.py#L1085
3. does this mean we 'ar', 'fr',... in the `context['lang']` are illegal in the future?

Reply:
Those few lines involving babel are actually only necessary when we are using `http.request.get_lang()`. The function gets the lang from the browser which isn't sanitized. (1) specifically `ar_AR` isn't known to babel hence we change it to `ar` and let babel find an alias it supports. (2) aliases (fr => fr_FR, en => en_US, ...) are only necessary when the lang is extracted from the browser, langs coming from the `res.lang` model have a full code already.

So indeed if we had kept that code, we should had changed `context['lang'] = 'ar'` to `lang = 'ar'` but that code is not necessary anymore as we don't take the lang from the request anymore, it is best to remove those 4 lines algother